firebase-firestore/src/main/java/com/google/firebase/firestore/core/RelationFilter.java

```diff
@@ -74,6 +77,17 @@ public boolean matches(Document doc) {
   private boolean matchesValue(FieldValue other) {
     if (operator == Operator.ARRAY_CONTAINS) {
       return other instanceof ArrayValue && ((ArrayValue) other).getInternalValue().contains(value);
+    } else if (operator == Operator.IN) {
+      return value instanceof ArrayValue && ((ArrayValue) value).getInternalValue().contains(other);
```

Comment:
This instanceof check on `value` is a little misleading because `value` will always be an `ArrayValue` (unless we have a bug in our query construction code), so I would rewrite this as:
```
hardAssert(value instanceof ArrayValue, "IN filter has invalid value: " + value);
if (((ArrayValue)value).getInternalValue()...
```